### PR TITLE
Travis tests

### PR DESCRIPTION
**Pull request by AHAAAAAAA**
Added Travis build checking

## Description
Checks Flask and Python versions. Installs all required modules. Runs the server with the -m -d -ts flags, terminates after 20 seconds if no errors and we get successful build. Rudimentary test, but a step in a good direction. First time I've integrated Travis, please review.

## Motivation and Context

Testing needs to be done, a lot of changes break our build and it would be nice to automate the test quickly.

## How Has This Been Tested?

Travis builds fine. Works as intended while not affecting other functionality.

## Screenshots (if appropriate):
screenshot 2016-07-25 13 00 32

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by AHAAAAAAA** [comment]
Addresses #1806, a starting point to go off.

**Comment by Chlodochar** [comment]
Yes! This is also useful for #2058. The tests aren't needed atm. The sleep doesn't do anything and `--test` is confusing for `runserver.py`. I'd say use a normal testing framework. From what I understand `nosetest` is an often-used framework for python. Travis yml and coupling Travis in general needs to be merged asap.

**Comment by AHAAAAAAA** [comment]
You suggest removing -ts option and the runserver script and replacing it with 'nosetest'?

**Comment by Chlodochar** [comment]
Yes, `runserver.py -ts` is of no use for the application itself. It's only for developers. I'd say follow the standards of Python development and use a testing framework if you want to add tests.

**Comment by Chlodochar** [comment]
That said, adding Travis YML is very useful. Others can iterate on that in their own PRs once the YML is in.

**Comment by AHAAAAAAA** [comment]
Updated with suggested fixes.

**Comment by Chlodochar** [comment]
The files look good to me :+1:. Let's see what Travis thinks :D

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+sudo: false
+language: python
+python:
+  - 2.7
+env:
+  - FLASK=0.11.1
+install:
+  - pip install Flask==$FLASK
```
Comment (chuyskywalker) [inline .travis.yml]: Why is this here? Doesn't `reuirements.txt` do this?

**Comment by Chlodochar** [comment]
Yay, Travis thinks it's good. :+1: for merging!

File: .travis.yml (hunk 1 above)
Comment (AHAAAAAAA) [inline .travis.yml]: Seemed to be a standard way of testing from all the .travis.yml files I've compared it against, but it seems redundant to me as well.

Comment (Chlodochar) [inline .travis.yml]: I'm guessing it can be installed with multiple versions when you expand the `env:` section. That way the library will be tested with multiple versions of Flask. Not really needed for this. So you can remove the env: section as well if you're going to remove the `pip install Flask` line.

**Comment by chuyskywalker** [comment]
> Ran 0 tests in 0.288s

Slick. :+1:

**Comment by Chlodochar** [comment]
Nice! :D :tada:

**Comment by Chlodochar** [comment]
added python linting for travis: #2079

File: .travis.yml (hunk 1 above)
Comment (efiku) [inline .travis.yml]: https://docs.travis-ci.com/user/customizing-the-build/

**Comment by efiku** [comment]
@AHAAAAAAA It's awesome change :D !
Now people can add TDD for backend functions.
For example  create stub around PokemonGo response (if it is possible), so Travis can test app with fake responses.

**Comment by Chlodochar** [comment]
This probably also needs to be added to pgoapi. @tjado what do you think? Sign up for Travis and Landscape.io, add them to your repo and add something similar to [.travis.yml](https://github.com/AHAAAAAAA/PokemonGo-Map/blob/develop/.travis.yml) too.

**Comment by Chlodochar** [comment]
Ping @tejado

**Comment by tejado** [comment]
Woa yeah, I also thought about tests for pgoapi. I have to find some time for this :o

**Comment by Chlodochar** [comment]
Landscape.io is almost zero-effort ;). It seems to be working: https://github.com/AHAAAAAAA/PokemonGo-Map/pull/2159/commits/5372ded791a7bde54aa26bcf3a2058d2ff7d02c5

**Comment by tejado** [comment]
My travis jobs are failing because it can't install protobuff... https://travis-ci.org/tejado/pgoapi/jobs/147432997 oO
